### ISSUE TITLE
Wrong results for TD-DFT with range-separated functionals and MEM_DF

### ISSUE THREAD

**Issue by hokru** [psi4/psi4#2431]
Original report: http://forum.psicode.org/t/tdscf-calculation-for-large-system/2407

Contrary to the forum report I found no issue with B3LYP.
Only the combination of range-separated functionals and `mem_df` seems to give wrong results.
Note, for current PSI4 v1.5 the `wcombine` option is turned off https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libfock/jk.cc#L111

H2O2 example:
```
from psi4.driver.procrouting.response.scf_response import tdscf_excitations
mol = geometry("""
   0  1
         O            0.000000000000     0.695000000000    -0.049338363732
         O            0.000000000000    -0.695000000000    -0.049338363732
         H           -0.388142000000     0.895249000000     0.783035636268
         H            0.388142000000    -0.895249000000     0.783035636268
symmetry c1
""")
psi4.set_options({
    'basis':'def2-SVP',
    'scf_type':'mem_df',
    'save_jk': True,
    'tdscf_tda':True,
})
e, wfn = psi4.energy("cam-b3lyp", return_wfn=True, molecule=mol)
res = tdscf_excitations(wfn, states=2,tda=True,r_convergence=2e-7)
```

[H2O2_memDF.txt](https://github.com/psi4/psi4/files/8027476/H2O2_memDF.txt)
[H2O2_PK.txt](https://github.com/psi4/psi4/files/8027475/H2O2_PK.txt)
[H2O2_diskDF.txt](https://github.com/psi4/psi4/files/8027477/H2O2_diskDF.txt)

---

QCHEM results
```
Excited state   1: excitation energy (eV) =    6.5611
Excited state   2: excitation energy (eV) =    6.7734
```

**Comment by JonathonMisiewicz** [comment]
> Contrary to the forum report I found no issue with B3LYP.

This is an important point, and I'd suggest we compare notes with the original reporter and figure out if this is reproducible or not before spending much time on this.

**Comment by susilehtola** [comment]
> > Contrary to the forum report I found no issue with B3LYP.
> 
> This is an important point, and I'd suggest we compare notes with the original reporter and figure out if this is reproducible or not before spending much time on this.

Right: there may be another issue that arises even for B3LYP when you get to the large system... Let's hope not

**Comment by JonathonMisiewicz** [comment]
Because I can't follow my own advice, I did some digging and found a hacky fix to the bug:
```
     if (lr_symmetric_) {
         if (do_wK_) {
             for (size_t N = 0; N < wK_ao_.size(); N++) {
                 wK_ao_[N]->hermitivitize();
             }
         }
+    } else {
+        if (do_wK_) {
+            for (size_t N = 0; N < wK_ao_.size(); N++) {
+                wK_ao_[N]->transpose_this();
+            }
+        }
+    }
```

**The `wK_ao_` matrices are currently the transpose of the correct values: and this causes the bug. Upon getting the entries in the correct order, MemDFJK and DiskDFJK agree.** In the standard SCF procedure, only the symmetric case of the JK engines run, and in that case, we forcibly hermitivitize the matrix. But in TD-HF, we are in the asymmetric case, so we cannot assume hermitian symmetry. Transposes matter.

_Why_ `MemDFJK` is wrong by a transpose is unclear, but based on the call stack, my money is a bug in `DFHelper::build_JK`.

**Comment by JonathonMisiewicz** [comment]
The bug is now fixed in master Psi. Expect a v1.5.1 where this issue is also fixed.

The reporter who brought this issue to our attention also claimed an issue in B3LYP (which we have tried and failed to reproduce) but did not provide a complete example of what they did to produce an issue, so the report may be in error. If the report can be verified, that will belong in a separate issue. This issue is very specifically about range-separated functionals.
